src/windows/models/properties_model.py

```diff
@@ -166,92 +155,113 @@ def update_frame(self, frame_number, reload_model=True):
+        if not i_id or not i_type:
+            return None
+        try:
+            timeline = get_app().window.timeline_sync.timeline
+            if i_type == "clip":
+                return timeline.GetClip(i_id)
+            if i_type == "transition":
+                return timeline.GetEffect(i_id)
+            if i_type == "effect":
+                return timeline.GetClipEffect(i_id)
+            raise RuntimeError("Unknown item type")
+        except Exception:
+            log.warning("Exception in get_lib_object", exc_info=1)
+            return None
+
+    @staticmethod
+    def get_query_object(q_id: str, q_type: str) -> QueryObject:
+        """Find the query object for a timeline item, by ID"""
+        if q_type == "clip":
+            return Clip.get(id=q_id)
+        if q_type == "transition":
+            return Transition.get(id=q_id)
+        if q_type == "effect":
+            return Effect.get(id=q_id)
+        return None
+
+    @staticmethod
+    def make_dataref(c, obj=None) -> dict:
+        try:
+            if obj:
+                return c.data.get('objects', {}).get(obj)
+            else:
+                return c.data
+        except Exception:
+            log.error("Couldn't interpret query object", exc_info=1)
+            return {}
+
+    @staticmethod
+    def optimize_data(c, prop_key, prop_data, obj=None):
+        """Reduce a query object to just the updated part"""
+        try:
+            if obj:
+                c.data = {'objects': {obj: {prop_key: prop_data}}}
+            else:
+                c.data = {prop_key: prop_data}
+        except Exception:
+            log.error("Failed to update query object", exc_info=1)
+
     def remove_keyframe(self, item):
         """Remove an existing keyframe (if any)"""
 
         # Determine what was changed
-        property = self.model.item(item.row(), 0).data()
-        property_type = property[1]["type"]
-        closest_point_x = property[1]["closest_point_x"]
-        property_type = property[1]["type"]
-        property_key = property[0]
-        object_id = property[1]["object_id"]
-        clip_id, item_type = item.data()
+        prop_key, prop_data = self.model.item(item.row(), 0).data()
+        prop_type = prop_data.get("type")
+        closest_x = prop_data.get("closest_point_x")
+        prop_type = prop_data.get("type")
+        object_id = prop_data.get("object_id")
+        item_id, item_type = item.data()
 
         # Find this clip
-        c = None
-        clip_updated = False
-
-        if item_type == "clip":
-            # Get clip object
-            c = Clip.get(id=clip_id)
-        elif item_type == "transition":
-            # Get transition object
-            c = Transition.get(id=clip_id)
-        elif item_type == "effect":
-            # Get effect object
-            c = Effect.get(id=clip_id)
-
-        if not c:
+        c = c = PropertiesModel.get_query_object(item_id, item_type)
```

Comment:
`c = c =` ??

Reply:
Ick, that's not supposed to be there!

Reply:
Fixed, and I couldn't resist sneaking in one more `@staticmethod` helper while I was at it.